Doc/library/code.rst

```diff
@@ -23,27 +23,33 @@ build applications which provide an interactive interpreter prompt.
    ``'__doc__'`` set to ``None``.
 
 
-.. class:: InteractiveConsole(locals=None, filename="<console>")
+.. class:: InteractiveConsole(locals=None, filename="<console>", block_exit=False)
```

Comment:
Can you come up with a better name? To me, `allow_exit=True` or even just `exit=True` feels better.

Reply:
So the difference between the option on and off is whether `exit()` would terminate the whole process, or just the interactive console. By `block_exit` I was trying to describe the option "blocks" the `exit()` function from exiting the whole process. `exit()` will always be allowed in the interactive console and it will always have effect. The only difference is to what level.

I would be happy to rename it to something better, but `allow_exit` probably does not describe the behavior well.

Reply:
Yeah, this name is tricky. Maybe one of the other reviewers has an idea? If not, we can go with block_exit.

Reply:
Yeah. Maybe `ignore_exit=False`?

When I see `block` here, it makes me think of a syntactic block (noun), or blocking (verb, as in "a blocking call"). It doesn't suggest "catch SystemExit" to me.

Reply:
Well it's not "ignoring" `exit()`, like I said above, `exit()` has effect, just different when the option is toggled. How about something like `local_exit`? Meaning we will only "exit" the interpreter.

Reply:
> Yeah. Maybe `ignore_exit=False`?
> 
> When I see `block` here, it makes me think of a syntactic block (noun), or blocking (verb, as in "a blocking call"). It doesn't suggest "catch SystemExit" to me.

Circling back to this as I still think this PR has some value. Let's figure out a name for it. Do you like `use_local_exit` or `local_exit`? Or maybe a vague description like `safe_exit`? Or we can label its actual mechanism - `exit_without_closing_stdin` - that's the key thing we need - the existing behavior also raises `SystemExit` but `stdin` is closed so we are basically done.